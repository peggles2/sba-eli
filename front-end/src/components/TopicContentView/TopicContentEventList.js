import React, { Component } from "react";
import { Icon, Grid } from "semantic-ui-react";
import axios from "axios";
import { Link } from "react-router-dom";

export default class TopicContentEventList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      eventsList: []
    };
  }

  componentDidMount() {
    this.eventsList();
  }

  eventsList() {
    const url = process.env.REACT_APP_SERVICE_HOST + `/learning_events/`;

    const eventParams = {
      course_id: this.props.course_id,
      module_id: this.props.module_id
    };

    axios
      .get(url, { params: eventParams })
      .then(res => {
        const eventsList = res.data;
        this.setState({ eventsList });
      })
      .catch(error => {
        console.log(error);
      });
  }

  renderEventList(eventList) {
    const { course_id, module_id } = this.props;
    const url = `/learning_paths/${course_id}/learning_objectives/${module_id}/learning_events/`;

    if (eventList && eventList.length) {
      return eventList.map(event => {
        return (
          <Grid
            className={"topic-content-event-grid"}
            key={"topicContentItem" + event.id}
          >
            <Grid.Column
              mobile={4}
              tablet={3}
              computer={2}
              largeScreen={1}
              widescreen={1}
            >
              <Icon
                name={"image"}
                className={"topic-content-event-grid-icon"}
              />
            </Grid.Column>
            <Grid.Column
              mobile={12}
              tablet={13}
              computer={14}
              largeScreen={13}
              widescreen={15}
            >
              <Link
                className={"topic-content-event-grid-header"}
                to={url + event.id}
              >
                {event.title}
              </Link>
              <div className={"topic-content-event-grid-meta"}>Event Type</div>
            </Grid.Column>
          </Grid>
        );
      });
    } else {
      return null;
    }
  }

  render() {
    return this.renderEventList(this.state.eventsList);
  }
}
