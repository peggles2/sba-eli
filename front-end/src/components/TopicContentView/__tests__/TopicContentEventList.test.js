import React from "react";
import { shallow, mount } from "enzyme";
import { TopicContentEventList } from "../TopicContentEventList";
import { Grid } from "semantic-ui-react";

describe("TopicContentViewList", () => {
  it("should render a <Grid>", () => {
    let props = {
      course_id: 1,
      module_id: 1,
      learningEventsCollection: {},
      dispatch: () => {}
    };
    let wrapper = shallow(<TopicContentEventList {...props} />);
    expect(wrapper.find(Grid).exists()).toBe(false);

    props = {
      course_id: 1,
      module_id: 1,
      learningEventsCollection: { 1: { 1: [{ title: "Test Event" }] } },
      dispatch: () => {}
    };
    wrapper = shallow(<TopicContentEventList {...props} />);
    expect(wrapper.find(Grid).exists()).toBe(true);
  });
});
