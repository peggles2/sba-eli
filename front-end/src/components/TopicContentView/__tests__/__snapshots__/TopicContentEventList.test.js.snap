// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicContentViewList should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopicContentEventList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div>
        <ItemGroup
          divided={true}
        >
          <Item>
            <ItemImage
              size="tiny"
              src="https://via.placeholder.com/50"
            />
            <ItemContent>
              <ItemHeader>
                Learning Event Test Title
              </ItemHeader>
              <ItemMeta>
                No Data
              </ItemMeta>
            </ItemContent>
          </Item>
        </ItemGroup>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ItemGroup
          divided={true}
        >
          <Item>
            <ItemImage
              size="tiny"
              src="https://via.placeholder.com/50"
            />
            <ItemContent>
              <ItemHeader>
                Learning Event Test Title
              </ItemHeader>
              <ItemMeta>
                No Data
              </ItemMeta>
            </ItemContent>
          </Item>
        </ItemGroup>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Item>
            <ItemImage
              size="tiny"
              src="https://via.placeholder.com/50"
            />
            <ItemContent>
              <ItemHeader>
                Learning Event Test Title
              </ItemHeader>
              <ItemMeta>
                No Data
              </ItemMeta>
            </ItemContent>
          </Item>,
          "divided": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ItemImage
                size="tiny"
                src="https://via.placeholder.com/50"
              />,
              <ItemContent>
                <ItemHeader>
                  Learning Event Test Title
                </ItemHeader>
                <ItemMeta>
                  No Data
                </ItemMeta>
              </ItemContent>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "size": "tiny",
                "src": "https://via.placeholder.com/50",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <ItemHeader>
                    Learning Event Test Title
                  </ItemHeader>,
                  <ItemMeta>
                    No Data
                  </ItemMeta>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Learning Event Test Title",
                  },
                  "ref": null,
                  "rendered": "Learning Event Test Title",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "No Data",
                  },
                  "ref": null,
                  "rendered": "No Data",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div>
          <ItemGroup
            divided={true}
          >
            <Item>
              <ItemImage
                size="tiny"
                src="https://via.placeholder.com/50"
              />
              <ItemContent>
                <ItemHeader>
                  Learning Event Test Title
                </ItemHeader>
                <ItemMeta>
                  No Data
                </ItemMeta>
              </ItemContent>
            </Item>
          </ItemGroup>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ItemGroup
            divided={true}
          >
            <Item>
              <ItemImage
                size="tiny"
                src="https://via.placeholder.com/50"
              />
              <ItemContent>
                <ItemHeader>
                  Learning Event Test Title
                </ItemHeader>
                <ItemMeta>
                  No Data
                </ItemMeta>
              </ItemContent>
            </Item>
          </ItemGroup>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Item>
              <ItemImage
                size="tiny"
                src="https://via.placeholder.com/50"
              />
              <ItemContent>
                <ItemHeader>
                  Learning Event Test Title
                </ItemHeader>
                <ItemMeta>
                  No Data
                </ItemMeta>
              </ItemContent>
            </Item>,
            "divided": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ItemImage
                  size="tiny"
                  src="https://via.placeholder.com/50"
                />,
                <ItemContent>
                  <ItemHeader>
                    Learning Event Test Title
                  </ItemHeader>
                  <ItemMeta>
                    No Data
                  </ItemMeta>
                </ItemContent>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "size": "tiny",
                  "src": "https://via.placeholder.com/50",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <ItemHeader>
                      Learning Event Test Title
                    </ItemHeader>,
                    <ItemMeta>
                      No Data
                    </ItemMeta>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Learning Event Test Title",
                    },
                    "ref": null,
                    "rendered": "Learning Event Test Title",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "No Data",
                    },
                    "ref": null,
                    "rendered": "No Data",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
